sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -757,6 +768,93 @@ Exceptions:
 * Throws synchronously with error code `invalid` if the resulting dependency would
   lead to a cycle. This error is omitted when `property::graph::no_cycle_check` is set.
 
+|
+[source,c++]
+----
+node add_barrier(const std::vector<event> eventList = {});
+----
+
+|Adds a barrier to the graph.
+
+Preconditions:
+
+* This member function is only available when the `command_graph` state is
+  `graph_state::modifiable`.
+
+Parameters:
+
+* `eventList` - Zero or more events to wait for.
+  These events can be graph-limited events (from graph recording) 
+  or general sycl events (such as events external to the graph).
+  If no event are provided, the barrier waits for all the previous nodes 
+  of the graph to complete.
+
+Returns: The command-group function object node which has been added to the graph.
+
+|
+[source,c++]
+----
+node make_external_event(event& recordedEvent, event& externalEvent, bool updatable = false);
```

Comment:
Rather than creating a new external event, could we just make the existing `recordedEvent` object external in-place?

Reply:
Some backends don't record events by default (typically CUDA and HIP). So when users request an external event, we have to add a new graph node to record this event. 
In the current implementation, we use graph-limited events as references their associated nodes. 
If we only transform these events into external events that means we will only get a single reference for two nodes which does not seem to be a good idea IMO. Moreover, if users update the external event, should we use the update event to reference the graph node?

Reply:
Makes sense thanks for the explanation. I think calling this `add_external_event()` would make that link to CUDA/HIP having to add commands clearer.

Wondering if we need `recordedEvent`, or could we pass a list of dependent nodes like we do for the regular `add()` API to be the edges to this new command that creates an external event. If this command is adding a node, it should be clear what the dependencies of that node are.